pandas/_libs/algos.pyx

```diff
@@ -99,11 +99,14 @@ def is_lexsorted(list list_of_arrays):
     cdef int64_t **vecs = <int64_t**> malloc(nlevels * sizeof(int64_t*))
     for i in range(nlevels):
         arr = list_of_arrays[i]
+        if arr.dtype.name != 'int64':
```

Comment:
can u make this an assert instead

Reply:
If this is a private function (?), an assert is OK. Otherwise I prefer raising a ValueError or TypeError. Maybe I am missing something.

Reply:
this is private